src/lib.rs

```diff
@@ -175,14 +175,24 @@ impl Decode for CENNZnutV0 {
 }
 
 impl Validate for CENNZnutV0 {
-    fn validate(&self, module_name: &str, method_name: &str) -> Result<(), &'static str> {
+    fn validate(
+        &self,
+        module_name: &str,
+        method_name: &str,
+        args: &[PactType],
+    ) -> Result<(), &'static str> {
         let module = self
             .get_module(module_name)
-            .ok_or("CENNZnut does not grant permission for module")?;
-        module
+            .ok_or_else(|| "CENNZnut does not grant permission for module")?;
+        let method = module
             .get_method(method_name)
-            .map(|_| ())
-            .ok_or("CENNZnut does not grant permission for method")
+            .ok_or_else(|| "CENNZnut does not grant permission for method")?;
+        if let Some(contract) = method.get_pact() {
+            interpret(args, contract.data_table.as_ref(), &contract.bytecode)
+                .map(|_| ())
```

Comment:
Need to check the boolean `Ok(bool)` from this result as this says whether the pact was upheld or not, an error means the contract itself error-ed for some reason. see: https://github.com/cennznet/pact/blob/3d12b18a3095a919dae204a91c2520c4129e3902/src/interpreter/mod.rs#L32